templates/goshared/map.go

```diff
@@ -3,6 +3,10 @@ package goshared
 const mapTpl = `
 	{{ $f := .Field }}{{ $r := .Rules }}
 
+	{{if .Rules.GetOmitempty}}
```

Comment:
Same comment as for repeated fields

Reply:
Maps support many validations. Omitempty is totally valid for maps IMO.